core-web/libs/sdk/analytics/src/lib/react/hook/useAnalyticTracker.ts

```diff
@@ -0,0 +1,45 @@
+import { useContext } from 'react';
+
+import { AnalyticsTracker } from '../../shared/dot-content-analytics.model';
+import { isInsideEditor } from '../../shared/dot-content-analytics.utils';
+import DotContentAnalyticsContext from '../contexts/DotContentAnalyticsContext';
+
+/**
+ * Hook to track analytics events in the application
+ *
+ * @returns {AnalyticsTracker} Object with track method to send events
+ *
+ * @example
+ * ```tsx
+ * const { track } = useAnalyticsTracker();
+ *
+ * const handleClick = () => {
+ *   track("btn-click", {
+ *     title: "My Title",
+ *     buttonText: "Link to detail",
+ *     urlTitle: "my-page"
+ *   });
+ * };
+ *
+ * return (
+ *   <button onClick={handleClick}>
+ *     Click me
+ *   </button>
+ * );
+ * ```
+ */
+export const useAnalyticsTracker = (): AnalyticsTracker => {
+    const instance = useContext(DotContentAnalyticsContext);
+    const insideEditor = isInsideEditor();
+
```

Comment:
Should memoization be applied? If the track function or its dependencies are complex or costly to recreate on each render, consider using useCallback to memoize the track function.

const track = useCallback((eventName: string, payload = {}) => {
       if (typeof instance?.track === 'function' && !insideEditor) {
           instance.track(eventName, {
               ...payload,
               timestamp: new Date().toISOString()
           });
       }
   }, [instance, insideEditor]);

Reply:
I changed this.